src/components/welcome/MyAccounts/index.tsx

```diff
@@ -13,21 +13,31 @@ import AddIcon from '@/public/images/common/add.svg'
 import { AppRoutes } from '@/config/routes'
 import ConnectWalletButton from '@/components/common/ConnectWallet/ConnectWalletButton'
 import useWallet from '@/hooks/wallets/useWallet'
+import { useRouter } from 'next/router'
 
 const NO_SAFES_MESSAGE = "You don't have any Safe Accounts yet"
-import { useRouter } from 'next/router'
 
 type AccountsListProps = {
-  safes: SafeItems
+  safes: SafeItems | undefined
   onLinkClick?: () => void
 }
 const AccountsList = ({ safes, onLinkClick }: AccountsListProps) => {
-  const router = useRouter()
-  const ownedSafes = useMemo(() => safes.filter(({ isWatchlist }) => !isWatchlist), [safes])
-  const watchlistSafes = useMemo(() => safes.filter(({ isWatchlist }) => isWatchlist), [safes])
   const wallet = useWallet()
-  const trackingLabel =
-    router.pathname === AppRoutes.welcome.accounts ? OVERVIEW_LABELS.login_page : OVERVIEW_LABELS.sidebar
+  const router = useRouter()
+
+  const ownedSafes = useMemo(() => safes?.filter(({ isWatchlist }) => !isWatchlist), [safes])
+  const watchlistSafes = useMemo(() => safes?.filter(({ isWatchlist }) => isWatchlist), [safes])
+
+  const isLoginPage = router.pathname === AppRoutes.welcome.accounts
+  const trackingLabel = isLoginPage ? OVERVIEW_LABELS.login_page : OVERVIEW_LABELS.sidebar
+
+  useEffect(() => {
+    // track owned and watchlist safe counts once when the user logs in
+    if (watchlistSafes && ownedSafes && isLoginPage) {
+      trackEvent({ ...OVERVIEW_EVENTS.TOTAL_SAFES_OWNED, label: ownedSafes.length })
+      trackEvent({ ...OVERVIEW_EVENTS.TOTAL_SAFES_WATCHLIST, label: watchlistSafes.length })
+    }
+  }, [isLoginPage, ownedSafes, watchlistSafes])
```

Comment:
This effect will run as many times as any of these dependencies change, right? I thought @alicecha wanted to emit this only one per hard page load.

Reply:
true it shouldn't send the events when you add a safe to the watchlist or anything like that. I removed ownedSafes watchlistSafes from the dependency array

Reply:
but isn't this now always going to send the safes when you navigate to the welcome page? I can click a safe, navigate to it, then hit back and then click on another safe.
In your first iteration it was when "on login page and watched safes and owned safes" change. Now you would always send it when on login page.
I'm not sure what we want here, but the way I understood is - once per session. What do we treat as a session? I think - when connected with walletX, if that is the case we need this somewhere higher in the component hierarchy?

Reply:
yes so it sends the events once on visiting the login page. The idea being that the frequency people usually visit the login page is about the frequency @alicecha wanted these data tracked. Tracking on wallet connect could work also but it is missing out on the case where the user uses safes in localstorage without connecting a wallet

Reply:
We need to think about what purpose this data will serve. Based on recent conversations, we want to know the number of Safes in the list(s) so we can
1) decide how to display them properly in the UI
2) understand how many users will be impacted by design choices, and how often (e.g. how many users needed to do an extra click when we started showing only 5 Safes)
3) estimate the impact of increasing the number of CGW requests.

Others??

Based on this, not sure whether it makes more sense to trigger the event every time a list is loaded rather than once per session. Wdyt?

Reply:
to clarify, by every time list is loaded you mean emit for:
- opening the sidebar.
- viewing accounts page
- adding or creating a safe.
If that's the case my guess is that it would be a lot of events with mostly the same data. Personally I think once per session would probably be enough to get an idea, but it's up to you @alicecha. We can also pare it back later if it's too much.

Reply:
Thanks for clarifying. Let's emit once per session!